### PR TITLE
Add: init npm-build-push

### PR DESCRIPTION
**Pull request by pascalholthaus**
## What
Add: init npm-build-push

## Why
DEVOPS-612

## References
DEVOPS-612

## Checklist
Not tested yet.

### CODE REVIEW

File: npm-build-push/action.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    - name: Setup Node.js
+      if: inputs.version
+      uses: actions/setup-node@v3
+      with:
+          node-version: ${{ inputs.version }}
+          registry-url: ${{ inputs.registry-url }}
+
+    - name: Run npm ci
+      shell: bash
+      run: npm ci
```
Comment (y0urself) [inline npm-build-push/action.yaml:32]:
```suggestion
    - name: Run npm ci
      shell: bash
      run: npm c
      if: ${{ inputs.run-ci }}
```

I do not think, that the CI is required for this step? Maybe you could make this an optional step?

File: npm-build-push/action.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    - name: Build npm packages
+      if: inputs.packages
+      shell: bash
+      run: |
+        # Set IFS to space only
+        IFS=$' '
+        # Build all pkgs from the space separated string inputs.packages
+        for pkg in ${{ inputs.packages }} ; do
+          npm run "$pkg"
+        done
```
Comment (y0urself) [inline npm-build-push/action.yaml:43]: Actually IIUC the `npm run` cmd runs predefined tasks specified in the `package.json`, e.g. this: https://github.com/greenbone/ui/blob/main/package.json#L11
We maybe should only use a `build-args` parameter and pass it? 
E.g. when using the `ui` repo you want to go with `npm run build` and pass `build` to that input parameter?

Comment (pascalholthaus) [reply npm-build-push/action.yaml:43]: You think of renaming inputs.packages to inputs.scripts ?

File: npm-build-push/action.yaml (hunk 1 above)
Comment (pascalholthaus) [reply npm-build-push/action.yaml]: done

**Comment by pascalholthaus** [comment]
i changed the default input script to build.

File: npm-build-push/action.yaml (hunk 2 above)
Comment (pascalholthaus) [reply npm-build-push/action.yaml]: I changed the default input script to build.

**Review by y0urself — APPROVED** [review]

**Pull request merged**